core/src/main/java/com/datastax/oss/driver/api/core/MappedAsyncPagingIterable.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core;
+
+import java.util.function.Function;
+
+/** The result of calling {@link #map(Function)} on another async iterable. */
+public interface MappedAsyncPagingIterable<ElementT>
```

Comment:
It's used to materialize `AsyncPagingIterable`'s self-type in calling code. Consider this for example:

```java
session
    .executeAsync("SELECT * FROM foo")
    .thenAccept(
        rs -> {
          MappedAsyncPagingIterable<String> mapped = rs.map(Object::toString);
          CompletionStage<MappedAsyncPagingIterable<String>> mapped2Future =
              mapped.fetchNextPage();
        });
```

If we want `map` to return `AsyncPagingIterable`, I think its signature would have to be:
```java
<TargetT, ResultT extends AsyncPagingIterable<TargetT, ResultT>> ResultT map(
      Function<? super ElementT, ? extends TargetT> elementMapper)
```

But then try to rewrite the first example.

Reply:
Ok, I see that it would be hard to use by the client because he would need to use whole self-type when assigning `mapped` to a variable. Thx for clarification